From: SeongJae Park <sj@kernel.org>
Date: Mon, 6 Oct 2025 22:56:00 -0700
Subject: [PATCH] mm/damon/core: fix wrong totalram usage on
 damos_get_node_memcg_used_bp()

The totalram value is in pages, but the code assumes it is in bytes.  As
a result, the goal score calculation is significantly wrong.  Fix the
wrong usage of the metric.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 6466efa215c7..65da79451bdb 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -2150,9 +2150,9 @@ static unsigned long damos_get_node_memcg_used_bp(
 
 	si_meminfo_node(&i, goal->nid);
 	if (goal->metric == DAMOS_QUOTA_NODE_MEMCG_USED_BP)
-		numerator = used_pages * PAGE_SIZE;
+		numerator = used_pages;
 	else	/* DAMOS_QUOTA_NODE_MEMCG_FREE_BP */
-		numerator = i.totalram - used_pages * PAGE_SIZE;
+		numerator = i.totalram - used_pages;
 	return numerator * 10000 / i.totalram;
 }
 #else
-- 
2.39.5

